src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,38 +1,13 @@
-package core.mate.academy.service;
-
-import java.util.List;
-
-/**
- * Parameterize this service and add its implementation in a separate class.
- */
-public interface MachineService {
-
-    /**
-     * Return the list of machines.
-     * In the implementation of this method please use your MachineProducer implementations
-     * See that 'Class type' is not parametrized.
-     * Consider to parametrize this Class< PARAMETRIZE_ME > with specific wildcard bounding
-     *
-     * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
-     * @return the list of machines
-     */
-    List<Object> getAll(Class type);
-
-    /**
-     * Fill the machines list with passed value
-     * Replace the Object with parametrized value.
-     * This method should be able to work well with any type of machine passed as 'value'
-     *
-     * @param machines - list of machines to be filled with value
-     * @param value    - any object of machine sub class
-     */
-    void fill(List<Object> machines, Object value);
-
-    /**
-     * Call the method doWork() from each machine.
-     * This method should be able to accept a list of bulldozers as well as list of trucks.
-     *
-     * @param machines - the list of machines
-     */
-    void startWorking(List<Object> machines);
-}
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import java.util.List;
+
+public interface MachineService<T extends Machine> {
+
+    List<T> getAll(Class<? extends T> type);
+
+    <T extends Machine> void fill(List<? super T> list, T value);

```

Comment:
We need just change Object type where it specified  to other type
```suggestion
    void fill(List<? super T> list, T value);

```

Reply:
Oookay